secret-rotator/svckey/svckey.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package svckey
+
+// package svckey implements the provisioning and deactivation of gcloud service account keys
+
+import (
+	"fmt"
+	"k8s.io/klog"
+)
+
+type ServiceAccountKeySpec struct {
+	Project        string `yaml:"project"`
+	ServiceAccount string `yaml:"serviceAccount"`
+}
+
+func (svc ServiceAccountKeySpec) String() string {
+	return fmt.Sprintf("projects/%s/serviceAccounts/%s", svc.Project, svc.ServiceAccount)
+}
+
+// ServiceAccountKeySpec.Type() is used to obtain the provisioner of the ServiceAccountKey
+func (svc ServiceAccountKeySpec) Type() string {
+	return "serviceAccountKey"
+}
+
+type Provisioner struct{}
+
+// TODO: add project and service-account fields to metadata when registered
+
+// CreateNew provisions a new service account key,
+// returns the key-id and private-key data of the created key if successful,
+// otherwise returns error
+func (p *Provisioner) CreateNew(labels map[string]string) (string, []byte, error) {
+	// TODO: provision new service account key
```

Comment:
would like to see this

Reply:
Implemented, will include in next commit.